server/context.go

```diff
@@ -60,43 +63,24 @@ func (s *SessionManager) NewSession(conn *mysql.Conn) {
 }
 
 // NewContext creates a new context for the session at the given conn.
-func (s *SessionManager) NewContext(conn *mysql.Conn) (*sql.Context, DoneFunc, error) {
-	ctx, cancel := context.WithCancel(context.Background())
+func (s *SessionManager) NewContext(conn *mysql.Conn) *sql.Context {
 	s.mu.Lock()
 	sess := s.sessions[conn.ConnectionID]
 	s.mu.Unlock()
-	context := sql.NewContext(ctx, sql.WithSession(sess), sql.WithTracer(s.tracer))
-	id := uuid.NewV4()
-
-	s.mu.Lock()
-	s.sessionContexts[conn.ConnectionID] = append(s.sessionContexts[conn.ConnectionID], id)
-	s.contexts[id] = cancel
-	s.mu.Unlock()
-
-	return context, func() {
-		s.mu.Lock()
-		defer s.mu.Unlock()
+	context := sql.NewContext(
+		context.Background(),
+		sql.WithSession(sess),
+		sql.WithTracer(s.tracer),
+		sql.WithPid(s.nextPid()),
```

Comment:
maybe pid can be atomic - wdyt?
to avoid in the future situations that `nextPid()` will block because mutex wasn't unlocked.
People are used to write:
```
lock()
defer unlock()
```
if we accidentally do sth. like that here, we will stuck in `nextPid`

Reply:
I mean, it could be atomic. But it's a given that we will unlock all mutexes at all times 😛 
We already use the mutex for other things, anyway, so the mutex will still be used even if it's not for this.